saleor/graphql/product/mutations/products.py

```diff
@@ -1700,8 +1701,19 @@ class Meta:
     @classmethod
     def perform_mutation(cls, _root, info, **data):
         media_obj = cls.get_node_or_error(info, data.get("id"), only_type=ProductMedia)
+        if media_obj.to_remove:
+            raise ValidationError(
+                {
+                    "media_id": ValidationError(
+                        "Media not found.",
+                        code=ProductErrorCode.NOT_FOUND,
+                    )
+                }
+            )
         media_id = media_obj.id
-        media_obj.delete()
+        media_obj.to_remove = True
+        media_obj.save(update_fields=["to_remove"])
+        delete_product_media_task.delay(media_id)
```

Comment:
I would run the task only if `ProductMedia` exists. Basically move line this line from task here `product_media = ProductMedia.objects.filter(pk=media_id, to_remove=True).first()`

Reply:
AFAIK we cannot pass media object to the celery task.

Reply:
Yes, just like @IKarbowiak mentioned. That is why I passed media id media can be fetched by task itself